### PR TITLE
[Feature-3754][Alert] Alert module support service for other service invoke

### PR DESCRIPTION
**Pull request by zhuangchong**
#3754

## What is the purpose of the pull request

In this feature , Alert server is a message notification service , email and wechat is a plugin of this service . If you want send email message you can create an instance of email plugin in the alert group and then you can select the alert group in SqlTask . SqlTask can call the alert server api with this alert group arrguement to send the message.

So we need alert api to support other service invoke.

Now the alert service sends alarms abnormally by periodically scanning the records in table t_ds_alert. It is asynchronous.

In the SqlTask task case, the task need send message synchronous and the final running status of this task depends on whether the message is sent successfully. So the alert api need send messages synchronously.

the params of alert api :

* alert group id : requied, One or more alert plugin instances are associated with an alert group, and these alert plugin instances will be called in turn when sending a message.
* title : requied, the title of this alert message.
* content: requied, the content of this alert message.

### CODE REVIEW

File: dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java [hunk 1]
```diff
@@ -118,10 +139,34 @@ public void start() {
...
+    /**
+     * stop
+     */
+    public void stop() {
+        this.server.close();
+        logger.info("alert server shut down");
+    }
+
     public static void main(String[] args) {
         System.out.println(System.getProperty("user.dir"));
```
Comment (CalvinKirs) [inline dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java:161]: Change to log output

Comment (zhuangchong) [reply dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java:161]: System.out.println is not clear who added it, so I did not move.

**Review by gaojun2048 — APPROVED** [review]
+1

**Review by CalvinKirs — APPROVED** [review]
+1

**Pull request merged**